Core/DISET/AuthManager.py

```diff
@@ -256,6 +262,26 @@ def getUsername(self, credDict):
       return True
     return False
 
+  def isUserSuspended(self, credDict):
+    """ Discover if the use is in Suspended status
+
+    :param credDict: Credentials to ckeck
+    :return: Boolean True if user is Suspended
+    """
+    if self.KW_USERNAME not in credDict:
+      self.getUsername(credDict)
+    if self.KW_USERNAME in credDict:
+      username = credDict[self.KW_USERNAME]
+    else:
+      return False
+    if self.KW_GROUP not in credDict:
+      return False
+    vo = CS.getVOForGroup(credDict[self.KW_GROUP])
+    suspendedVOList = CS.getUserOption(username, 'Suspended', [])
+    if vo in suspendedVOList:
+      return True
+    return False
```

Comment:
The method can be a bit more efficient, and efficiency in this specific case is important:

```
    if self.KW_USERNAME not in credDict:
      self.getUsername(credDict)
    if self.KW_USERNAME not in credDict or self.KW_GROUP not in credDict:
      return False
    suspendedVOList = CS.getUserOption(credDict[self.KW_USERNAME], 'Suspended', [])
    if not suspendedVOList:
      return False
    vo = CS.getVOForGroup(credDict[self.KW_GROUP])
    if vo in suspendedVOList:
      return True
    return False

```

Reply:
Replaced